### ISSUE TITLE
Deploy before/after getting reset

### ISSUE THREAD

**Issue by bramdevries** [rocketeers/rocketeer#396]
I was having trouble adding a `deploy.after` task and after some debugging I found the following.

Basically I have a file `tasks.php` where I do the following:

```php

Rocketeer::listenTo('deploy.before-symlink', [
	Grunt::class,
]);

Rocketeer::listenTo('deploy.after', [
	Restart::class
]);
```

The `deploy-before-symlink` works fine, but for some reason the `deploy.after` was never triggered. I found that this was caused because the `TaskHandler::registerConfiguredEvents` was being called twice.  This resets all tasks for a certain set of events (`deploy.after` being one of them).

If I add the task in the `hooks.php` file it gets registered properly. It's probably caused because the `tasks.php` file gets loaded before Rocketeer is properly booted, thus the tasks being reset. I thought it was an interesting issue to mention since I've seen you use the `tasks.php` file too.

**Comment by Anahkiasen** [comment]
Hm can't seem to reproduce this, tried on a fresh app, if I add your exact tasks and then dump the listeners I do get the correct output:

```
Macedoine ➜  sandbox-standalone  rocketeer tinker
[1] rocketeer> $rocketeer['rocketeer.tasks'];
// object(Rocketeer\Services\TasksHandler)(
//
// )
[2] rocketeer> $rocketeer['rocketeer.tasks']->getTasksListeners('deploy', 'after', true);
// array(
//   0 => 'restart'
// )
[3] rocketeer> $rocketeer['rocketeer.tasks']->getTasksListeners('deploy', 'before-symlink', true);
// array(
//   0 => 'grunt'
// )
```

Do you have anything in your hooks file?

**Comment by Schlaefer** [comment]
I see the same behaviour if I use the `--on="foo"` flag and `foo` is not the the default connection in `config.php`. `deploy.after` is not triggered then.

In `ConnectionsHandler::setConnection` the expression `$this->connection == $connection` is false and doesn't return early, so `TaskHandler::registerConfiguredEvents` is called a second time.

**Comment by bramdevries** [comment]
@Schlaefer Yes I'm using stages too which I guess triggers the `registerConfiguredEvents` and resets the hooks.

**Comment by Anahkiasen** [comment]
Should work now on develop, can you check?

**Comment by bramdevries** [comment]
It works, but only if I don't include the `RocketeerSlackServiceProvider`, seems to get reset somewhere else then.

**Comment by Schlaefer** [comment]
A quick test of develop throws me nested error #398, so I can't verify at the moment.

**Comment by Anahkiasen** [comment]
Damn, that was the easy solution, if that doesn't work now I need to work on the hard one >_<
